server.go

```diff
@@ -114,6 +116,17 @@ func NewServer() *Server {
 		shardsBySeriesID: make(map[uint32][]*Shard),
 		Logger:           log.New(os.Stderr, "[server] ", log.LstdFlags),
 	}
+	// Server will always return with authentication enabled.
+	// To set the server to 'authless mode', call server.SetAuthenticationEnabled(false)
+	s.authenticationEnabled = true
+	return &s
+}
+
+// SetAuthenticationEnabled turns on or off server authentication
+func (s *Server) SetAuthenticationEnabled(enabled bool) {
+	s.mu.Lock()
```

Comment:
I mean, the underlying instruction to set an int to 0 or 1 (or however it is represented) is going to be a single CPU instruction, right? It is not interrruptible.